### ISSUE TITLE
React Native Metro server with Babel 7 configuration crashes if the project has images

### ISSUE THREAD

**Issue by ajcrites** [facebook/metro#245]
I'm not sure if this is really a metro issue or not or if the underlying issue needs to be handled elsewhere. I had initially reported this here: https://github.com/react-navigation/react-navigation/issues/4983 -- but the issue will happen if babel finds any image files (and possibly other non-source files) including images used in the app.

From what I can tell, setup for babel 7 is not well documented. I will describe my setup in detail below, but I cobbled it together from https://github.com/facebook/react-native/issues/19859#issuecomment-407748189

The simplest way to set this up from scratch:

1. `npx react-native-cli@latest init project`
2. Remove `.babelrc`. Replace with: `babel.config.js`:

```
module.exports = function (api) {
  api.cache(false);
  return {
    presets: ["module:metro-react-native-babel-preset"],
    plugins: ["@babel/transform-runtime"],
  }
};
```

3. `yarn add --dev @babel/plugin-transform-runtime @babel/runtime`
4. `yarn react-native bundle --entry-file index.js --bundle-output does-not-matter`
5. Add an image to the project. For example, copy any image and have `<Image source={require('path to your image')} />` in App.js

This happens whether or not you have `@babel/core` and the bridge installed.

**What is the current behavior?**
With a babel 7 setup, if your project source has any images at all including images in third party libraries that are transpiled, `react-native start` will not build the app:

```
error: bundling failed: TypeError: comments is not iterable
    at PluginPass.Program (test-project/node_modules/@babel/plugin-transform-flow-strip-types/lib/index.js:58:31)
    at newFn (test-project/node_modules/@babel/traverse/lib/visitors.js:193:21)
    at NodePath._call (test-project/node_modules/@babel/traverse/lib/path/context.js:53:20)
    at NodePath.call (test-project/node_modules/@babel/traverse/lib/path/context.js:40:17)
    at NodePath.visit (test-project/node_modules/@babel/traverse/lib/path/context.js:88:12)
    at TraversalContext.visitQueue (test-project/node_modules/@babel/traverse/lib/context.js:118:16)
    at TraversalContext.visitSingle (test-project/node_modules/@babel/traverse/lib/context.js:90:19)
    at TraversalContext.visit (test-project/node_modules/@babel/traverse/lib/context.js:146:19)
    at Function.traverse.node (test-project/node_modules/@babel/traverse/lib/index.js:94:17)
```

The actual issue occurs here: https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-flow-strip-types/src/index.js#L28 because `comments` is `null`. After doing an `if (comments)` wrapper, another similar error occurs babel-plugin-transform-react-jsx.

**If the current behavior is a bug, please provide the steps to reproduce and a minimal repository on GitHub that we can `yarn install` and `yarn test`.**
Example repo: https://github.com/ajcrites/metro-crash-image-build -- you can reproduce the issue via `yarn react-native start` and `yarn react-native run-ios`, or `yarn react-native bundle`.

**What is the expected behavior?**
The project should build without any issues.

**Please provide your exact Metro configuration and mention your Metro, node, yarn/npm version and operating system.**
* node v10.10
* yarn v1.9
* react-native v0.57 (can be reproduced with `npx react-native-cli@latest init`)
* metro-react-native-babel-preset v0.45.2

**Comment by ajcrites** [comment]
You can currently work around this by adding `exclude: ['**/*.png', '**/*.jpg', '**/*.gif', ...]` to the top level of the configuration object created in `babel.config.js`.

**Comment by rafeca** [comment]
Thanks for reporting!

The issue comes by babel v7 still reading `babel.config.js` files when Metro is doing some internal transformations. Up until now Metro has been using `babelrc: false` to disable the babel automatic resolution of `.babelrc` but it turns out that now with Babel 7 we need to do the same with config files ([more info](https://babeljs.io/docs/en/next/config-files)). We're going to fix this soon and release a patch version of Metro to fix this in React Native 0.57
